src/Microsoft.Extensions.AwaitableStream/AwaitableStream.cs

```diff
@@ -37,9 +41,11 @@ public class AwaitableStream : Stream
 
         public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            var cancellation = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _cancellationTokenSource.Token).Token;
```

Comment:
Yeah, realized that and fixed in UnpooledChannel version. This won't be needed if we go with that.